### PR TITLE
[ZA-17] Add support for multi-host apps

### PR DESCRIPTION
**Pull request by danielbreves**
:koala:

/cc @zendesk/quokka

### References
 - Jira link: https://zendesk.atlassian.net/browse/ZA-17

### Risks
 - App locations may not validate correctly

### CODE REVIEW

**Comment by sandlerr** [comment]
is it possible to persist the conversion of the locations array into a hash? That way at least everything that gets uploaded from now on will follow our new schema

File: spec/validations/manifest_spec.rb [hunk 1]
```diff
@@ -95,6 +95,21 @@ def create_package(parameter_hash)
...
+  it 'should not have an error when the app host is valid' do
+    manifest = { 'location' => { 'zopim' => ['chat_sidebar'] } }
+    allow(@manifest).to receive_messages(read: JSON.generate(manifest))
+
+    expect(@package).not_to have_error(/invalid host/)
+    expect(@package).not_to have_error(/invalid location/)
+  end
+
+  it 'should have an error when the app host is invalid' do
+    manifest = { 'location' => { 'freshdesk' => ['ticket_sidebar'] } }
```
Comment (adammw) [inline spec/validations/manifest_spec.rb]: :laughing:

**Comment by adammw** [comment]
:+1:

**Comment by danielbreves** [comment]
@sandlerr I will, but this will be in ZAM, ZAS only provides validations.

**Comment by sandlerr** [comment]
:+1:

**Comment by sandlerr** [comment]
@danielbreves this PR changed `en.yml`, does it also need to touch https://github.com/zendesk/zendesk_apps_support/blob/master/config/locales/translations/zendesk_apps_support.yml ?

**Pull request merged**